scIB/metrics.py

```diff
@@ -558,97 +561,61 @@ def cell_cycle(adata_pre, adata_post, batch_key, embed=None, agg_func=np.mean,
     scores_before = []
     scores_after = []
-    #if both (s-score, g2m-score) and pc-regression are pre-computed 
-    if (np.in1d(['S_score', 'G2M_score'], 
-                adata_pre.obs_keys()).sum() == 2) and ('scores_before' in adata_pre.uns_keys()): 
-        #extract needed infos from adata_pre and delete it from memory
-        df_pre = adata_pre.obs[['S_score', 'G2M_score', batch_key]]
-        
-        scores_precomp = pd.Series(adata_pre.uns['scores_before'])
-        del adata_pre
-        n_item = gc.collect()
-        
-        for batch in enumerate(batches):
-            raw_sub = df_pre.loc[df_pre[batch_key] == batch[1]]
-            int_sub = adata_post[adata_post.obs[batch_key] == batch[1]].copy()
-            int_sub = int_sub.obsm[embed] if embed is not None else int_sub.X
-        
-            if raw_sub.shape[0] != int_sub.shape[0]:
-                message = f'batch "{batch[1]}" of batch_key "{batch_key}" '
-                message += 'has unequal number of entries before and after integration.'
-                message += f'before: {raw_sub.shape[0]} after: {int_sub.shape[0]}'
-                raise ValueError(message)
-        
-            if verbose:
-                print("score cell cycle")
-            
-            covariate = raw_sub[['S_score', 'G2M_score']]
-            after =  pc_regression(int_sub, covariate, pca_var=None, n_comps=n_comps, verbose=verbose)
-            scores_after.append(after)
-            #get score before from list of pre-computed scores
-            before = scores_precomp[batch[1]]
-            scores_before.append(before)
-
-            score = 1 - abs(after - before)/before # scaled result
-            if score < 0:
-                # Here variance contribution becomes more than twice as large as before
-                if verbose:
-                    print("Variance contrib more than twice as large after integration.")
-                    print("Setting score to 0.")
-                score = 0
-        
-            scores_final.append(score)
-        
-            if verbose:
-                print(f"batch: {batch[1]}\t before: {before}\t after: {after}\t score: {score}")
-                 
-    else: #not everything is pre-computed
-       
-        for batch in batches:
-            raw_sub = adata_pre[adata_pre.obs[batch_key] == batch]
-            int_sub = adata_post[adata_post.obs[batch_key] == batch]
-            int_sub = int_sub.obsm[embed] if embed is not None else int_sub.X
-        
-            if raw_sub.shape[0] != int_sub.shape[0]:
-                message = f'batch "{batch}" of batch_key "{batch_key}" '
-                message += 'has unequal number of observations before and after integration.'
-                message += f'before: {raw_sub.shape[0]} after: {int_sub.shape[0]}'
-                raise ValueError(message)
-        
+
+    recompute_cc = recompute_cc or ('S_score' not in adata_pre.obs_keys() and 'G2M_score' not in adata_pre.obs_keys())
```

Comment:
should this not be an `or` in there? We still can't compute scores if only one of `'S_score'` or `'G2M_score'` are there.

Reply:
True, then it'd better be
```py
recompute_cc = recompute_cc or 'S_score' not in adata_pre.obs_keys() or 'G2M_score' not in adata_pre.obs_keys()
```